src/components/util/BaseEvent.ts

```diff
@@ -0,0 +1,61 @@
+
+
+export default abstract class BaseEvent {
```

Comment:
How about renaming this to something more descriptive?  Maybe `AbstractUserActionEvent`?